lib/charms/opensearch/v0/opensearch_config.py

```diff
@@ -171,6 +173,19 @@ def set_node(
             True,
         )
 
+    def add_seed_hosts(self, cm_ips: List[str]):
+        """Add CM nodes ips / host names to the seed host list of this unit."""
+        cm_ips_hostnames = cm_ips.copy()
+        for ip in cm_ips:
+            name, aliases, addresses = socket.gethostbyaddr(ip)
+            cm_ips_hostnames.extend([name] + aliases + addresses)
+
+        with open(self._opensearch.paths.seed_hosts, "w+") as f:
+            existing = [line.strip() for line in f.readlines()]
```

Comment:
do we need an existing list? I mean existing list can contain nodes that changed IP or nodes that was removed

Reply:
Hmm, I though it wouldn't hurt - since as long as some of these entries are up-and-running no matter if some are down.
We can remove it and see how it behaves